src/utils.h

```diff
@@ -578,14 +578,14 @@ inline int peek_next_nonws(std::istream& in) {
     *_p = '\0';                                         \
   }
 
-inline string to_hex(unsigned int * message_digest, const int len = 1)
+inline string to_hex(unsigned char * message_digest, const int len = 4)
```

Comment:
```suggestion
#if BOOST_VERSION >= 108600
inline string to_hex(unsigned char * message_digest, const int len = 4)
#else
inline string to_hex(unsigned int * message_digest, const int len = 1)
#endif
```

Reply:
Changed to use `boost::uuids::detail::sha1::digest_type` for `message_digest`. And changed the default value of `len` to be the size of `boost::uuids::detail::sha1::digest_type` in nibbles. `len` now controls the number of hex characters in the digest string.